plugins/woocommerce/includes/admin/importers/class-wc-product-csv-importer-controller.php

```diff
@@ -155,6 +138,19 @@ protected static function check_file_path( string $path ): void {
 		}
 	}
 
+	/**
+	 * Check if a given file is inside a given directory.
+	 *
+	 * @param string $file_path The full path of the file to check.
+	 * @param string $directory The path of the directory to check.
+	 * @return bool True if the file is inside the directory.
+	 */
+	private static function file_is_in_directory( string $file_path, string $directory ): bool {
+		$file_path = (string) new URL( $file_path ); // This resolves '/../' sequences.
+		$file_path = preg_replace( '/^file:\\/\\//', '', $file_path );
+		return 0 === stripos( wp_normalize_path( $file_path ), trailingslashit( wp_normalize_path( $directory ) ) );
```

Comment:
(Optional/feel free to skip) this would also work:

```suggestion
		$file_path = $file_path->get_path();
		return 0 === stripos( wp_normalize_path( $file_path ), trailingslashit( wp_normalize_path( $directory ) ) );
```